File: vital-items/src/main/java/me/vitalframework/items/VitalNamespacedKey.java
package me.vitalframework.items;

import org.bukkit.NamespacedKey;

/**
 * Defines namespaced keys for various purposes within the Vital plugin.
 * This interface centralizes key definitions to avoid conflicts and ensure consistency.
 *
 * @author xRa1ny
 */
public interface VitalNamespacedKey {
    /**
     * A namespaced key for storing a unique identifier on ItemStacks.
     */
    NamespacedKey ITEM_UUID = NamespacedKey.fromString("item-uuid");
}

File: vital-items/src/main/kotlin/me/vitalframework/items/VitalNamespacedKey.kt
package me.vitalframework.items

import org.bukkit.NamespacedKey

/**
 * Internal class; stores internal [NamespacedKey] instances for [VitalItem] and localization via `vital-localization`.
 */
object VitalNamespacedKey {
    @JvmStatic
    val ITEM_UUID = NamespacedKey("vital", "item-uuid")

    @JvmStatic
    val ITEM_LOCALIZED = NamespacedKey("vital", "item-localized")

    @JvmStatic
    val ITEM_LOCALIZATION_KEY = NamespacedKey("vital", "item-localization-key")

    @JvmStatic
    val ITEM_LORE_LOCALIZATION_KEYS = NamespacedKey("vital", "item-lore-localization-keys")
}
